ksqldb-common/src/main/java/io/confluent/ksql/logging/processing/ProcessingLoggerFactoryImpl.java

```diff
@@ -18,36 +18,123 @@
 import io.confluent.common.logging.StructuredLoggerFactory;
 import java.util.Collection;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.function.BiFunction;
+import java.util.function.Function;
+
+import io.confluent.ksql.util.MetricsTagsUtil;
+import org.apache.kafka.common.MetricName;
+import org.apache.kafka.common.metrics.Metrics;
+import org.apache.kafka.common.metrics.Sensor;
+import org.apache.kafka.common.metrics.stats.CumulativeSum;
 
 public class ProcessingLoggerFactoryImpl implements ProcessingLoggerFactory {
+  public static final String PROCESSING_LOG_ERROR_METRIC_NAME = "processing-error-total";
+  public static final String PROCESSING_LOG_METRICS_GROUP_NAME = "processing-diagnostic-metrics";
+  public static final String PROCESSING_LOG_METRIC_DESCRIPTION =
+      "The total number of errors emitted by the processing log.";
+
   private final ProcessingLogConfig config;
   private final StructuredLoggerFactory innerFactory;
+  private final Metrics metrics;
+  private final Map<String, String> metricsTags;
   private final BiFunction<ProcessingLogConfig, StructuredLogger, ProcessingLogger> loggerFactory;
+  private final Function<Metrics, BiFunction<ProcessingLogger, Sensor, ProcessingLogger>>
+      loggerWithMetricsFactory;
+  private final Map<String, ProcessingLogger> processingLoggers;
 
   ProcessingLoggerFactoryImpl(
       final ProcessingLogConfig config,
-      final StructuredLoggerFactory innerFactory) {
-    this(config, innerFactory, ProcessingLoggerImpl::new);
+      final StructuredLoggerFactory innerFactory,
+      final Metrics metrics,
+      final Map<String, String> metricsTags
+  ) {
+    this(
+        config,
+        innerFactory,
+        metrics,
+        ProcessingLoggerImpl::new,
+        metricObject -> (processingLogger, sensor) ->
+            new MeteredProcessingLogger(processingLogger, metricObject, sensor),
+        metricsTags
+    );
   }
 
   ProcessingLoggerFactoryImpl(
       final ProcessingLogConfig config,
       final StructuredLoggerFactory innerFactory,
-      final BiFunction<ProcessingLogConfig, StructuredLogger, ProcessingLogger> loggerFactory
+      final Metrics metrics,
+      final BiFunction<ProcessingLogConfig, StructuredLogger, ProcessingLogger> loggerFactory,
+      final Function<Metrics, BiFunction<ProcessingLogger, Sensor, ProcessingLogger>>
+          loggerWithMetricsFactory,
+      final Map<String, String> metricsTags
   ) {
     this.config = config;
     this.innerFactory = innerFactory;
+    this.metrics = metrics;
     this.loggerFactory = loggerFactory;
+    this.loggerWithMetricsFactory = loggerWithMetricsFactory;
+    this.metricsTags = metricsTags;
+    this.processingLoggers = new ConcurrentHashMap<>();
   }
 
   @Override
-  public ProcessingLogger getLogger(final String name) {
-    return loggerFactory.apply(config, innerFactory.getLogger(name));
+  public ProcessingLogger getLoggerWithMetrics(
```

Comment:
nit: I don't think these names need the `withMetrics` suffix

Reply:
Should we just change the interface for the factory from `public interface ProcessingLoggerFactory` to `public interface MeteredProcessingLoggerFactory`? Then we could drop the with metrics suffix. I think somewhere in the interface, it should indicate that these loggers are emitting metrics.